### PR TITLE
added Edwards curves for MNT-298 and MNT-753

### PR DESCRIPTION
**Pull request by weikengchen**
With the help of Pratyush, we found Edwards curve parameters for MNT-298 and MNT-753. The former was easy. The latter took ~$1k AWS credits together with half a month of two machines in RISELab.

Comments are made extensive to aid others in case they need to construct an Edwards curve and want to know how to properly write the parameters in Zexe.

### CODE REVIEW

**Comment by Pratyush** [comment]
Thanks for the PR @weikengchen ! I just merged a different PR #213 which slightly changes the convention regarding naming of curves; I'd appreciate if you could modify the PR to be in line with the new convention:
* `edwards_xxx` -> `ed_on_xxx`.

File: algebra/src/edwards_mnt4_298/fields/fr.rs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    const TWO_ADICITY: u32 = 1u32;
+
+    // ROOT_OF_UNITY = GENERATOR ^ t =
+    // 118980571542315331438337312413262112886281219744507561120271964887686106682370032123932630
+    // t is defined below
+    // This number needs to be in the Montgomery residue form.
+    // I.e., write
+    // 118980571542315331438337312413262112886281219744507561120271964887686106682370032123932630
+    // * R
+    // =14596494758349247937872919467301196219547084259323651055171406111196152579418790325693086
```
Comment (Pratyush) [inline algebra/src/edwards_mnt4_298/fields/fr.rs]:
```suggestion
    // = 14596494758349247937872919467301196219547084259323651055171406111196152579418790325693086
```

File: algebra/src/edwards_mnt4_298/fields/fr.rs [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        4913018085921565342u64,
+        18164325898792356216u64,
+        11499902056485864693u64,
+        12113224729248979119u64,
+        126057789046u64,
+    ]);
+}
+impl FpParameters for FrParameters {
+    // MODULUS = 118980571542315331438337312413262112886281219744507561120271964887686106682370032123932631
+    // Factors:
```
Comment (Pratyush) [inline algebra/src/edwards_mnt4_298/fields/fr.rs]:
```suggestion
    // Factors of MODULUS - 1:
```

File: algebra/src/edwards_mnt4_753/curves/mod.rs [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+        16943442107464u64,
+    ]));
+
+    /// Generated randomly
+    const AFFINE_GENERATOR_COEFFS: (Self::BaseField, Self::BaseField) = (GENERATOR_X, GENERATOR_Y);
+
+    type MontgomeryModelParameters = EdwardsParameters;
+
+    /// Multiplication by `a` is just negation.
+    /// Is `a` 1 or -1?
```
Comment (Pratyush) [inline algebra/src/edwards_mnt4_753/curves/mod.rs]:
```suggestion
```

File: algebra/src/edwards_mnt4_753/fields/fr.rs [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+        5393472405610595666u64,
+        9815530206026813666u64,
+        9111703519527971535u64,
+        9880873531994141485u64,
+        37593433148184u64,
+    ]);
+}
+impl FpParameters for FrParameters {
+    // MODULUS = 5237311370989869175293026848905079641021338739994243633972937865128169101571388346632361720473792365177258871486054600656048925740061347509722287043067341250552640264308621296888446513816907173362124418513727200975392177480577
+    // Factors:
```
Comment (Pratyush) [inline algebra/src/edwards_mnt4_753/fields/fr.rs]:
```suggestion
    // Factors of MODULUS - 1:
```

**Comment by weikengchen** [comment]
no problem. will do the changes. will push later tonight

**Comment by weikengchen** [comment]
I made the changes (but seems CI would not work unless conflicts are resolved, which I tried but failed from my end)

**Comment by weikengchen** [comment]
aha, let me merge with the upstream in my repo and then make a new one.

**Pull request closed without merging**